src/components/Comment.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { formatDistance, subDays } from 'date-fns';
+import CommentForm from './CommentForm.tsx';
+
+type CommentTypes = {
+  id: string;
+  username: string;
+  image: string;
+  body: string;
+  userId: string;
+  createdAt: string;
+};
+type ActiveComment = {
+  id: string;
+  type: string; // You can specify more specific types for 'type' if needed
+};
+type CommentProps = {
+  comment: CommentTypes;
+  replies: Array<CommentTypes>;
+  setActiveComment: (comment: ActiveComment | null) => void;
+  activeComment: ActiveComment | null;
+  updateComment: (text: string, id: string) => void;
+  deleteComment: (id: string) => void;
+  addComment: (text: string, replyId: string) => void;
```

Comment:
❌ 
these are all event emitters, so the name should start with `on` 

`onUpdateComment` 
`onDeleteComment` etc